lib/src/local_working_copy.rs

```diff
@@ -763,8 +765,40 @@ fn mtime_from_metadata(metadata: &Metadata) -> MillisSinceEpoch {
     )
 }
 
+/// On Windows, checks if a path is a reparse point (junction or symlink).
+/// Junctions have is_dir()=true but is_symlink()=false, so we need this
+/// separate check to avoid recursing into them and hitting "Access denied".
+#[cfg(windows)]
+fn is_reparse_point(metadata: &Metadata) -> bool {
+    const FILE_ATTRIBUTE_REPARSE_POINT: u32 = 0x400;
+    (metadata.file_attributes() & FILE_ATTRIBUTE_REPARSE_POINT) != 0
+}
+
+#[cfg(windows)]
+fn is_normal_dir(entry: &DirEntry) -> bool {
+    entry.file_type().map(|ft| ft.is_dir()).unwrap_or(false)
+        && std::fs::symlink_metadata(entry.path())
+            .map(|m| !is_reparse_point(&m))
+            .unwrap_or(false)
+}
+
+#[cfg(not(windows))]
+fn is_normal_dir(entry: &DirEntry) -> bool {
+    entry.file_type().map(|ft| ft.is_dir()).unwrap_or(false)
+}
+
 fn file_state(metadata: &Metadata) -> Option<FileState> {
     let metadata_file_type = metadata.file_type();
+    // On Windows, junctions have is_symlink()=true but is_dir()=false (in Rust),
+    // even though they are directories. We want to ignore them.
+    #[cfg(windows)]
+    if metadata.file_attributes() & 0x10 != 0 {
+        return None;
+    }
+
+    // On Windows, junctions (created by tools like pnpm) have is_dir()=true
+    // but is_symlink()=false. They cannot be read as symlinks or regular files.
+    // Returning None causes jj to skip them as "special files".
```

Comment:
Does these mean there are at least two types of "junction"? Could you elaborate?

iirc, there are legacy "junction" that was available in Windows NT 3.5 or 2000 era, and newer symbolic link (which is more similar to Unix ones.) My understanding is that the latter should be mapped to symbolic link entry in Git, and this patch makes the former be ignored.